### PR TITLE
Added Sinhala Language (si)

### PR DESCRIPTION
**Pull request by SuwanSankaja**
Added Sinhala Readme file (si)

- added a new language support in Localization folder. 
- the new language in Sinhala (si)

Pull request description

- Created a new folder named in Localization folder 'si' and added a README file with the necessary details in Sinhala

> For detailed contributing instructions see https://github.com/iluwatar/java-design-patterns/wiki/01.-How-to-contribute

### CODE REVIEW

**Review by SuwanSankaja — COMMENTED** [review]
added sinhala languages

File: localization/si/README.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
+<!-- පහත පේළිය හිස් රේඛාවක් විය යුතුය C: (එය kramdown නොවන බැවිනි
+     ශීර්ෂයකට පෙර හිස් රේඛාවක් අවශ්‍ය බව බුද්ධිමත් හා ආදරණීයයි
+     එය එසේ පෙන්වන්න, උදා. වෙබ් අඩවිය) -->
```
Comment (iluwatar) [inline localization/si/README.md:3]: This metadata part needs to be in English. See for example here: https://raw.githubusercontent.com/iluwatar/java-design-patterns/master/localization/es/abstract-document/README.md

**Review by iluwatar — CHANGES_REQUESTED** [review]

File: localization/si/README.md (hunk 1 above)
Comment (SuwanSankaja) [reply localization/si/README.md]: changed to English.

**Review by iluwatar — APPROVED** [review]

**Comment by iluwatar** [comment]
Looks good! Thank you for the contribution 🎉

@all-contributors please add @SuwanSankaja for translation

**Pull request merged**